blaze-client/src/main/scala/org/http4s/client/blaze/Http1Connection.scala

```diff
@@ -86,11 +91,22 @@ private final class Http1Connection(val requestKey: RequestKey,
   }
 
   @tailrec
-  def reset(): Unit = {
+  private def reset(): Unit = {
     stageState.get() match {
       case v@ (Running | Idle) =>
         if (stageState.compareAndSet(v, Idle)) parser.reset()
         else reset()
+      case t @ Tunneling(sslStage) =>
+        sendOutboundCommand(Command.Disconnect)
+        // TODO why doesn't this work?  The next request gets a bunch of ^U
```

Comment:
Okay, if we're tunneling and keeping a connection alive, I think the proper model is to CONNECT once, and then put the connection into the pool for that request key.  Trying to peel back the SSLEngine only to reconnect again doesn't make any sense.